### DISCUSSION TITLE
Using the fetch() function of Workers service binding as a const

### DISCUSSION THREAD

**Discussion by nicougou** [cloudflare/workerd#2055] [Q&A] [answered]
Hi !
Did anyone ever manage to achieve something similar to this, using service binding in node ?

```
const fetchFn = env.MY_SERIVE_BIND?.fetch ?? fetch;
await fetch('url', {})
```

I am always getting `[ERROR] TypeError: Illegal invocation`.

I would have thought this could be possible, the goal is to be able to pass the `fetch()` function to other clients.

I tried using `env.MY_SERIVE_BIND?.fetch.bind(globalThis)` but it didn't help.

**Comment by kentonv** [comment, marked as answer]
To turn a method into a stand-alone function, you need to bind the method to the object you got it from, not to `globalThis`:

`env.MY_SERVICE_BIND.fetch.bind(env.MY_SERVICE_BIND)`

Note that this is generally true of all methods in JavaScript, not specific to `fetch` or to service bindings.

**Reply by nicougou** [reply]
@kentonv that was incredibly helpful, thanks !

While this was an obvious lack of knowledge on my end, this could be very helpful in the official docs, replacing the fetch client in some sdk that allows it by a service binding can help a lot :)
